backend/src/services/auto_trader.py

```
@@ -33,238 +27,195 @@ def __init__(self, socketio=None, data_fetcher_service=None, ai_analysis_service
-                logger.error(f"AutoTrader: Failed to execute emergency sell for {self.owned_tokens[token_address]['symbol']}: {sell_result.get('message')}")
-                if self.socketio:
-                    self.socketio.emit('auto_trade_event', {'type': 'sell', 'token': self.owned_tokens[token_address]['symbol'], 'reason': 'rugpull_cut_loss', 'status': 'failed', 'error': sell_result.get('message')})
-            
-            del self.owned_tokens[token_address] # Remove from owned tokens after attempting to sell
-        else:
-            logger.info(f"AutoTrader: Rugpull alert for unowned token {token_address}. No action taken.")
-
-# Create a singleton instance
-# auto_trader_service = AutoTraderService() # Instantiation will be handled in main.py
+            logger.warning(f"EMERGENCY SELL for {token_address} due to rugpull alert!")
+            details = self.owned_tokens[token_address]
```

Ruff (0.15.0)

[error] Local variable `details` is assigned to but never used

Remove assignment to unused variable `details`

(F841)